Add native types to ContactBundle entities and interfaces

Prokyonn opened the pull request:
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | no
| BC breaks? | yes
| Deprecations? | no
| License | MIT

#### What's in this PR?

Adds native types to entities and their methods.

#### Why?

Better developer experience.

Prokyonn committed:
Add native types to ContactBundle entities and interfaces

wachterjohannes commented on src/Sulu/Bundle/ContactBundle/Controller/PositionController.php:
```diff
@@ -177,6 +177,7 @@ public function cdeleteAction(Request $request)
                 }
 
                 $positionId = $position->getId();
+                \assert(null !== $positionId);
```
no asserts in controller

wachterjohannes commented on src/Sulu/Bundle/ContactBundle/Controller/PositionController.php:
```diff
@@ -215,6 +216,7 @@ public function deleteAction($id)
                 }
 
                 $positionId = $position->getId();
+                \assert(null !== $positionId);
```
no asserts in controller

wachterjohannes committed:
Fix ContactBundle native types issues

- Remove assertions from PositionController per code review
- Add null coalescing for firstName/lastName in trash restore
- Remove Collection type hints to prevent Doctrine proxy errors
- Add default values for non-nullable properties (firstName, lastName, name, lft, rgt, depth)
- Fix mainContact handling in AccountController to accept both int and array formats
- Change Account API mainContact serialization to return ID instead of full object

wachterjohannes committed:
Fix PositionController cpatchAction for Symfony type requirements

- Replace Request::get($i) with Request::toArray() to get JSON payload
- Change while loop to foreach loop for cleaner iteration
- Fixes TypeError with Symfony requiring string parameter for get()

This resolves the PositionControllerTest::testCpatchAction failure where
the PATCH request was failing due to passing an integer to Request::get()
instead of a string.
